terraform-plans/templates/github/snap_check.yaml.tftpl

```diff
@@ -135,7 +135,9 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: "3.10"
-
+%{ if extra_func_steps != "" }
+${indent(6, chomp(extra_func_steps))}
+%{ endif }
```

Comment:
I'm really not a fan of templating yaml within yaml... it's very prone to error.

Ideally we're wanting to keep the templates as consistent as possible, so having a freeform 'extra steps' field goes against that ideal. I'd like to avoid too many variables for the templates, especially variables that have large values.

How about a variable like `needs_juju = "true"`? Or just always installing juju? Or perhaps a separate template for these two projects that need juju?

Reply:
I see your point. I would like to avoid having two separate templates, because it's very easy to forget to sync the common parts if changes are needed.

I'll apply your suggestion of using `needs_juju`